enclave/core/sgx/calls.c

```diff
@@ -1255,6 +1248,52 @@ void oe_abort_with_td(oe_sgx_td_t* td)
             host_ecall_context->debug_eexit_rsp = frame[0] + 8;
             host_ecall_context->debug_eexit_rip = frame[1];
         }
+
+        // For debug enclaves, log the backtrace before marking the enclave as
+        // aborted.
+        {
+            // Fetch current values of FS and GS. Typically, FS[0] == FS and
+            // GS[0] == GS.
```

Comment:
This only holds after `td_init()` is called (the first-time _handle_ecall invocation). If an abort occurs prior to that both F[0] and G[0] will be zero, and the ocall cannot be made.

Reply:
Is there some way for us to detect that scenario? It is OK for now if we don't have bactrace prior to init.

Reply:
I've used a simple check pre td_init scenario. Let me know what you think.

Outside the scope of this PR, one thing to look at is if we can set up td early enough in oe_enter.
We already populate some fields in oe_enter: https://github.com/openenclave/openenclave/blob/60b2486b1a480852dd8ca83e90d245bb92ceb1fc/enclave/core/sgx/enter.S#L68. If we can populate some more fields to be able to make ocalls, then we can have ability to log at more places during enclave flow.

Another approach is to have a log-buffer passed in ecall_context. Then we could log to this buffer, and have the host emit it when it gets control.

Reply:
For this PR, I think skipping the following code when GS[0] equals to 0 is sufficient. Based on whether there is backtrace or not is good enough to determine if the abort occurs prior to td_init. We can have leave this as future work.